tour_model_eval/user_input_request_times_all_users.ipynb

```diff
@@ -0,0 +1,493 @@
+   "source": [
+    "radius = 100"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "lesbian-employee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "req_day = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "wound-thomson",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "req_month = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "arabic-growth",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "req_propor_median = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "formed-drive",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# build a base dataframe for request times in January \n",
+    "date_df = pd.DataFrame(data = {'date':np.arange(1,32),'drop_col':np.arange(1,32)})\n",
+    "date_df.set_index(['date'], inplace=True)\n",
+    "\n",
+    "# get valid user list\n",
+    "user_ls,valid_users = evaluation.get_user_ls(all_users,radius)\n",
+    "\n",
+    "for a in range(len(all_users)):\n",
+    "    user = all_users[a]\n",
+    "    filter_trips,sim,trips = evaluation.filter_data(user,radius)\n",
+    "    logging.debug(\"len(filter_trips)is %s \"% len(filter_trips))\n",
+    "\n",
+    "    # filter out users that don't have enough valid labeled trips\n",
+    "    if not evaluation.valid_user(filter_trips,trips):\n",
+    "        req_day.append(NaN)\n",
+    "        req_month.append(NaN) \n",
+    "        continue\n",
+    "    sim.bin_data()\n",
+    "    sim.delete_bins()\n",
+    "    bins = sim.bins\n",
+    "    \n",
+    "    # collect requested trips and common trips(no need to request) indices above cutoff\n",
+    "    ab_trip_ls = []\n",
+    "    no_req_trip_ls = []\n",
+    "    for bin in bins:\n",
+    "        early_trip_index, index = evaluation.find_first_trip(filter_trips,bin)\n",
+    "        ab_trip_ls.append(early_trip_index)\n",
+    "        \n",
+    "        for k in range(len(bin)):\n",
+    "            if k != index:\n",
+    "                no_req_trip_idx = bin[k]\n",
+    "                no_req_trip_ls.append(no_req_trip_idx)\n",
+    "\n",
+    "\n",
+    "    \n",
+    "    # bins below cutoff\n",
+    "    bl_bins = sim.below_cutoff\n",
+    "    \n",
+    "    # collect requested trips indices below cutoff\n",
+    "    bl_trip_ls = []\n",
+    "    for bin in bl_bins:\n",
+    "        for trip_index in bin:\n",
+    "            bl_trip_ls.append(trip_index)\n",
+    "            \n",
+    "    # whole list of requested trips indices\n",
+    "    req_trips_ls=ab_trip_ls+bl_trip_ls\n",
+    "    \n",
+    "    \n",
+    "    # collect request times in a day\n",
+    "    bin_day = evaluation.bin_date(req_trips_ls,filter_trips,day=True)\n",
+    "    req_day_ls = []\n",
+    "    for bin in bin_day:\n",
+    "        req_day_ls.append(len(bin))\n",
+    "               \n",
```

Comment:
Please add a comment here that describes what `bin_day` is expected to look like to make the rest of this easier to review